urlb44e9843ba_QuasimodoNZ_SWEN222_Game_Project_tgz-pJ8-Game_World_GameWorldLevelBuilderJ8/src/User_Interface/ApplicationWindow.java

```diff
@@ -0,0 +1,981 @@
+		textSP.setPreferredSize(new Dimension((toolBarHeight / 2),
+				toolBarHeight / 2));
+
+		// Change the text inside the chatArea
+		chatArea.setForeground(Color.black);
+		chatArea.setBackground(Color.white);
+		Font font = new Font("Verdana", Font.BOLD, 10);
+		chatArea.setFont(font);
+		chatArea.setLineWrap(true);
+		chatArea.setEditable(false);
+
+		// Setup a chatBox for where you type and send the chat
+		final JTextField chatBox = new JTextField(16);
+		chatBox.setPreferredSize(new Dimension((toolBarHeight / 2) + 20, 20));
+		chatBox.addActionListener(new ActionListener() {
+
+			@Override
+			public void actionPerformed(ActionEvent e) {
+				// Sends chat message to all the other charAreas
+				JTextField textField = (JTextField) e.getSource();
+				String text = textField.getText();
+				try {
+					// Sents text to the client
+					client.sendChat(name, text);
+					appendChatArea(name, text);
+
+					// Empties the chatBox
+					chatBox.setText("");
+
+					// Sets the focus to the frame for mouse clicks
+					frame.requestFocusInWindow();
+				} catch (IOException e1) {
+					return;
+				}
+			}
+
+		});
+
+		// Scrolls to the latest text all the time
+		DefaultCaret caret = (DefaultCaret) chatArea.getCaret();
+		caret.setUpdatePolicy(DefaultCaret.ALWAYS_UPDATE);
+
+		// Add an panel to hold all of the interaction buttons
+		interactionPanel = new JPanel();
+		TitledBorder title = BorderFactory
+				.createTitledBorder("Selected Object");
+		title.setTitleJustification(TitledBorder.CENTER);
+		interactionPanel.setBorder(title);
+		interactionPanel.setOpaque(false);
+		interactionPanel.setPreferredSize(new Dimension(200, 150));
+
+		// Add healthBar to healthPanel
+		healthPanel.add(Box.createRigidArea(new Dimension(0, 25)));
+		healthPanel.add(healthBar);
+
+		// Add components to the chatPanel
+		chatPanel.add(Box.createRigidArea(new Dimension(0, 40)));
+		chatPanel.add(textSP);
+		chatPanel.add(chatBox);
+
+		// Adds components to the toolbarContent
+		toolbarContent.add(healthPanel);
+		toolbarContent.add(inventoryPanel);
+		toolbarContent.add(chatPanel);
+		toolbarContent.add(interactionPanel);
+
+		// Adds the toolbar content to the toolbarPanel
+		toolbarPanel.add(toolbarContent);
+
+	}
+
+	/**
+	 * Append text to the chatArea
+	 *
+	 * @param Text
+	 *            to be appended
+	 */
+	public void appendChatArea(String name, String text) {
+		chatArea.append(name + ": " + text + "\n");
+	}
+
+	/**
+	 * Changes the interaction panel according to the selected object.
+	 * This method is called when an object is selected from either the renderer or the inventory.
+	 * Buttons are displayed in the panel depending on the type of the selected object.
+	 * A label of the selected object is also displayed
+	 * 
+	 * @param origin is equal to "Renderer" if the object was selected from the renderPanel
+	 * Or if it is equal to "Inventory" if it was from the inventoryPanel.
+	 */
+	public void setInteractionPanel(String origin) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 16. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.